### PR TITLE
Remove Julia 0.3 support

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

**Comment by andreasnoack** [comment]
:+1:

**Comment by simonbyrne** [comment]
This also removes `ArrayViews`. As `SubArray` isn't a subtype of `DenseArray`, I've added changed all occurrences to `StridedArray`.

**Comment by simonster** [comment]
+1000

File: src/genericrand.jl [hunk 1]
```diff
@@ -32,7 +32,7 @@ function rand!(s::Sampleable{Multivariate}, A::AbstractVector)
     _rand!(s, A)
 end
 
-function rand!(s::Sampleable{Multivariate}, A::DenseMatrix)
+function rand!(s::Sampleable{Multivariate}, A::StridedMatrix)
```
Comment (simonster) [inline src/genericrand.jl]: I'm getting an ambiguity warning from this:

```
WARNING: New definition 
    rand!(Distributions.Sampleable{Distributions.Multivariate, S<:Distributions.ValueSupport}, Union{DenseArray{T<:Any, 2}, Base.SubArray{T<:Any, 2, A<:DenseArray, I<:Tuple{Vararg{Union{Base.Range{Int64}, Int64, Base.Colon}}}, LD<:Any}}) at /home/simon/.julia/Distributions/src/genericrand.jl:36
is ambiguous with: 
    rand!(Distributions.Distribution{Distributions.Multivariate, S<:Distributions.ValueSupport}, AbstractArray{T<:Any, 2}) at /home/simon/.julia/Distributions/src/multivariates.jl:12.
To fix, define 
    rand!(Distributions.Distribution{Distributions.Multivariate, S<:Distributions.ValueSupport}, Union{DenseArray{T<:Any, 2}, Base.SubArray{T<:Any, 2, A<:DenseArray, I<:Tuple{Vararg{Union{Base.Range{Int64}, Int64, Base.Colon}}}, LD<:Any}})
before the new definition.
```

**Pull request merged**